upsertGitHubTag-f79c4f36-3e61-43f4-8f6c-0b2e0e9774d7/deployment/index.js

```diff
@@ -115,49 +115,79 @@ function processEvent(event, callback) {
-     * for expected fields. There are no enums for push events subtypes.
-     * 
-     * If an event is deemed not supported, we will return a success and print a message saying the event is not supported.
-     */
-    if (['repository', 'ref', 'created', 'deleted', 'pusher'].some(str => !(str in body))) {
-        console.log('Event is not supported')
-        callback(null, {"statusCode": 200, "body": "Currently, this lambda does not support this event type from GitHub."});
-    }
-
-    // A push has been made for some repository (ignore pushes that are deletes)
-    if (!body.deleted) {
-        console.log('Valid push event');
-        const repository = body.repository.full_name;
+    // Handle installation events
+    var githupEventType = event.headers["X-GitHub-Event"]
+    if (githubEventType === "installation_repositories") {
+        console.log('Valid installation event');
         const username = body.sender.login;
-        const gitReference = body.ref;
         const installationId = body.installation.id;
-        
-        var pushPostBody = {
-                "gitReference": gitReference,
-                "installationId": installationId,
-                "repository": repository,
-                "username": username
-            };
+        const repositoriesAdded = body.repositories_added;
+        var repositories = [];
+        repositoriesAdded.forEach((repo) => {
+            repositories.push(repo.full_name);
+        });
 
-        path += "workflows/github/release";
-        
+        var pushPostBody = {
+            "installationId": installationId,
+            "username": username,
+            "repositories": repositories.join(",")
+        };
+        path += "workflows/github/install";
+            
         postEndpoint(path, pushPostBody, (response) => {
-            const successMessage = 'The associated entries on Dockstore for repository ' + repository + ' with version ' + gitReference + ' have been updated';
+            const successMessage = 'The GitHub app was successfully installed on repository ' + repository;
             handleCallback(response, successMessage, callback);
         });
+    } else if (githubEventType === "push") {
+        /**
+         * We only handle push events, of which there are many subtypes. Unfortunately, the only way to differentiate between them is to look
+         * for expected fields. There are no enums for push events subtypes.
+         * 
+         * If an event is deemed not supported, we will return a success and print a message saying the event is not supported.
+         */
+        if (['repository', 'ref', 'created', 'deleted', 'pusher'].some(str => !(str in body))) {
+            console.log('Event is not supported')
+            callback(null, {"statusCode": 200, "body": "Currently, this lambda does not support this event type from GitHub."});
+        }
+
+        // A push has been made for some repository (ignore pushes that are deletes)
```

Comment:
You aren't ignoring pushes that are deletes, you're handling them in the else block, aren't you?

Reply:
Yes exactly. My comment is for the first if statement, though I can see how that is confusing.